### ISSUE TITLE
🐛 BUG: cloudflare/wrangler-action@2.0.0 fails to recognize branch

### ISSUE THREAD

**Issue by janithcooray** [cloudflare/workers-sdk#2569]
### Which Cloudflare product(s) does this pertain to?

Pages, Wrangler

### What version of `Wrangler` are you using?

2.0.0

### What operating system are you using?

Linux

### Describe the Bug

We’ve been using Wrangler CLI to deploy Cloudflare pages projects for more than 6 months now without a single issue.
we have over 35 pages projects with the exact same deployment strategy. this issue only occurred in 1 so far.

- there were no other open issues similar to this
- there were no format guidelines for issues so i wrote in my own format.

# BUG description 

we have more than 1 branch on each github repository containing pages source.  When we make a commit on a main branch, it gets deployed as a production build. when we make a commit on a branch other than main, it gets deployed as preview build. yesterday, we found that a commit made to non-production branch lead to a production build.

# Steps to reproduce (random occurrence)

1. Create a pages project
2. Create a repo on github and commit the following files to the main branch

- Create a basic html file under `app/public/index.html`
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Production Build</title>
</head>
<body>
    
</body>
</html>
```
- and the workflow file `.github/workflows/main.yml`
```
name: Deploy to Pages
on:
  push:
    branches: [ main, staging, dev/** ]
  pull_request:
    branches: [ staging ]

env:
  PROJECT_NAME: "<INSERT PAGES PROJECT NAME>"

jobs:
  deploy:
    runs-on: ubuntu-latest
    name: Deploy
    steps:
      - uses: actions/checkout@v3      
      - uses: cloudflare/wrangler-action@2.0.0
        with:
          apiToken: ${{ secrets.PAGES_DEPLOY_API }}
          accountId: ${{ secrets.PAGES_DEPLOY_ACCOUNT }}
          workingDirectory: 'app'
          command: pages publish public --project-name=${{ env.PROJECT_NAME }}
```
- (replace the project name to match the pages project and cloud credentials required by wrangler)

- github actions workflow will deploy the project to the pages account as a production build(takes a few seconds)

3. Checkout on a new branch (ex- named "staging") and make a simple change.
Checkout to a new branch and change the `Production Build` text in the html file to `Preview Build`

at this point, github actions workflow should deploy the project to the pages account as a preview build.

however, since yesterday preview builds are recognized as production build on the above deployment strategy within the cloudflare pages project

# Occurrence Frequency :
So far this only happened on 1 project, others with the same workflow did not have this bug.

# Environment Info
cloudflare/wrangler-action@2.0.0 on GitHub private repository
Runs on: Github actions - ubuntu-latest

**Comment by janithcooray** [comment]
# Update
# Occurrence Frequency

Seems to be happening on all projects now. cloudflare fails to recognise branches

**Comment by tomByrer** [comment]
@janithcooray current version is 2.**8**.0; did you misread that number (like I did also)?

**Comment by wallslide** [comment]
I had the exact same experience three days ago. I ended up hardcoding the branch name using `--branch` to allow me to deploy without overwriting my production site with our beta branch:
`pages publish dist --branch=beta  --project-name=***`

Haven't tried to deploy to a preview branch without that `--branch` flag since, so I'm not sure if it's still happening.

**Comment by wallslide** [comment]
This issue is still occurring. I am using this code to ensure the proper branch is credited:

`pages publish dist --branch=${{ github.head_ref || github.ref_name }} --project-name=***`

**Comment by janithcooray** [comment]
> @janithcooray current version is 2.**8**.0; did you misread that number (like I did also)?

nope its 2.0.0. its right there on the deployment file
```sh
      - uses: cloudflare/wrangler-action@2.0.0    
```

**Comment by janithcooray** [comment]
> This issue is still occurring. I am using this code to ensure the proper branch is credited:
> 
> `pages publish dist --branch=${{ github.head_ref || github.ref_name }} --project-name=***`

same, bug is over a week old. im a single DevOps engineer and i dont have enough manpower to go replacing all workflow files in all our pages repositories and reverse them after bug fixing. ill try to see if its an issue with wrangler or just Cloudflare pages.

**Comment by janithcooray** [comment]
could be an issue from action - https://github.com/cloudflare/wrangler-action

**Comment by janithcooray** [comment]
well fond the issue, more or less

using wrangler latest v2.8.1 solved the issue.

`cloudflare/wrangler-action@2.0.0` is not available but you can use `wranglerVersion` and set it to `2.8.1`

i made a pull request https://github.com/cloudflare/wrangler-action/pull/98 fixing this issue, also added `pagesBranch: true` which an additional work around. not sure if this will even get merged

**Comment by wallslide** [comment]
Unfortunately setting `wranglerVersion` to `2.8.1` didn't fix the issue for me.

**Comment by janithcooray** [comment]
> Unfortunately setting `wranglerVersion` to `2.8.1` didn't fix the issue for me.

this is maintained by me, try this and see if it works

```sh
env:
  PROJECT_NAME: <project-name>

jobs:
  deploy:
    runs-on: ubuntu-latest
    name: Deploy
    steps:
      - uses: actions/checkout@v3      
      - uses: 3CSDesign/wrangler-action@master
        with:
          apiToken: ${{ secrets.PAGES_DEPLOY_API }}
          accountId: ${{ secrets.PAGES_DEPLOY_ACCOUNT }}
          workingDirectory: 'app'
          pagesDirectory: 'public'
          pagesProject: ${{ env.PROJECT_NAME }}
```

**Comment by vimarshacooray** [comment]
Update. 

Now that worker round is broken at v2.8.1

setting --branch=preview is not working anymore

**Comment by janithcooray** [comment]
Use the latest version of Wrangler. it has a Deployment environment option to fix this.
No solution to fix 2.0.0

Closing since they are doing a rewrite - https://github.com/cloudflare/wrangler-action/discussions/111
